apps/create-fuels-counter-guide/src/components/Contract.tsx

```diff
@@ -0,0 +1,139 @@
+import { useEffect, useState } from "react";
+import { useWallet } from "@fuels/react";
+
+import LocalFaucet from "./LocalFaucet";
+import { TestContract } from "../sway-api";
+import Button from "./Button";
+import { isLocal, contractId, renderTransactionId } from "../lib.tsx";
+import { useNotification } from "../hooks/useNotification";
+
+export default function Contract() {
+  const { infoNotification, successNotification, errorNotification } =
+    useNotification();
+  const [contract, setContract] = useState<TestContract>();
+  const [counter, setCounter] = useState<number>();
+  const [isLoading, setIsLoading] = useState(false);
+
+  const { wallet, refetch } = useWallet();
+
+  useEffect(() => {
+    if (wallet) {
+      const testContract = new TestContract(contractId, wallet);
+      setContract(testContract);
+    }
+  }, [wallet]);
+
+  useEffect(() => {
+    if (contract && !counter) {
+      const getCount = async () => {
+        const { value } = await contract.functions.get_count().get();
+        setCounter(value.toNumber());
+      };
+
+      getCount();
+    }
+  }, [contract, counter]);
+
+  async function incrementCounter() {
+    if (!wallet || !contract) return;
+    setIsLoading(true);
+
+    try {
+      const call = await contract.functions.increment_counter(1).call();
+      infoNotification(<span>Transaction submitted: {renderTransactionId(call.transactionId)}</span>);
```

Comment:
I think this would cleaner if it lived in the notification hook, what do you think?

```ts
transactionSubmittedNotification(call.transactionId);
```

Same for a success notificaiton. Then the render function can live there as well rather than `lib`.

Reply:
I feel like the render function can still live in `lib` since that logic to render the link to the explorer can potentially be used elsewhere in the future